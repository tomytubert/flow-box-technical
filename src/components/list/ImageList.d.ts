import { Images } from '../../interfaces/types';
import PropTypes from 'prop-types';
interface Props {
    posts: Images[];
    className?: string;
}
interface CardProps {
    image: string;
    altImage: string;
    likes: number;
}
export declare const ImageList: {
    ({ posts, className }: Props): import("react/jsx-runtime").JSX.Element;
    propTypes: {
        posts: PropTypes.Validator<any[]>;
        className: PropTypes.Requireable<string>;
    };
};
export declare const CardList: {
    ({ image, altImage, likes }: CardProps): import("react/jsx-runtime").JSX.Element;
    propTypes: {
        image: PropTypes.Validator<string>;
        altImage: PropTypes.Validator<string>;
        likes: PropTypes.Validator<number>;
    };
};
export {};
