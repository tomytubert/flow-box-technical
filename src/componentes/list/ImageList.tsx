import { Box, Card, CardContent, CardMedia, Typography } from '@mui/material'
import { Images } from '../../interfaces/types'
import FavoriteIcon from '@mui/icons-material/Favorite'

interface Props {
	posts: Images[]
}

interface CardProps {
	image: string
	altImage: string
	likes: number
}

export const ImageList = ({ posts }: Props) => {
	return (
		<Box gap={2} display="flex" flexDirection="column">
			{posts.map((post) => {
				return (
					<CardList
						image={post.urls.full}
						altImage={post.alt_description}
						likes={post.likes}
					/>
				)
			})}
		</Box>
	)
}

const CardList = ({ image, altImage, likes }: CardProps) => {
	return (
		<Card sx={{ display: 'flex' }}>
			<CardMedia
				component='img'
				height='194'
				image={image}
				alt={altImage}
				sx={{ width: '200px', flex:'2' }}
				loading='lazy'
			/>
			<CardContent sx={{ height: '120px',flex:'3' }}>
				<Box display='flex' alignItems='center' gap={2} mb={1}>
					<FavoriteIcon />
					<Typography variant='body2' color='text.secondary'>
						{likes}
					</Typography>
				</Box>
				<Typography variant='body2' color='text.secondary'>
					{altImage}
				</Typography>
			</CardContent>
		</Card>
	)
}
