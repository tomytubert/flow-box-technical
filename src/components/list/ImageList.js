import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { Box, Card, CardContent, CardMedia, Typography } from '@mui/material';
import FavoriteIcon from '@mui/icons-material/Favorite';
import PropTypes from 'prop-types';
export var ImageList = function (_a) {
    var posts = _a.posts, className = _a.className;
    return (_jsx(Box, { gap: 2, display: 'flex', flexDirection: 'column', className: className, children: posts.map(function (post) {
            return (_jsx(CardList, { image: post.urls.full, altImage: post.alt_description, likes: post.likes }, post.id));
        }) }));
};
export var CardList = function (_a) {
    var image = _a.image, altImage = _a.altImage, likes = _a.likes;
    return (_jsxs(Card, { sx: { display: 'flex' }, children: [_jsx(CardMedia, { component: 'img', height: '194', width: '200', image: image, alt: altImage, loading: 'lazy', sx: { flex: { md: 2 }, width: '200px', minWidth: '200px' } }), _jsxs(CardContent, { sx: { height: { md: '120px' }, flex: { md: 2 } }, children: [_jsxs(Box, { display: 'flex', alignItems: 'center', gap: 2, mb: 1, children: [_jsx(FavoriteIcon, { color: 'primary' }), _jsx(Typography, { variant: 'body2', color: 'text.secondary', children: likes })] }), _jsx(Typography, { variant: 'body2', color: 'text.secondary', className: 'capitalize-first-letter', children: altImage })] })] }));
};
ImageList.propTypes = {
    posts: PropTypes.array.isRequired,
    className: PropTypes.string,
};
CardList.propTypes = {
    image: PropTypes.string.isRequired,
    altImage: PropTypes.string.isRequired,
    likes: PropTypes.number.isRequired,
};
