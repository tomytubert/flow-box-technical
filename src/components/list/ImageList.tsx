import { Box } from '@mui/material'
import { Images } from '../../interfaces/types'
import { CardPost } from '../card/Card'

interface ImageListProps {
	posts: Images[]
	className?: string
}

export const ImageList = ({ posts, className }: ImageListProps) => {
	return (
		<Box gap={2} display='flex' flexDirection='column' className={className}>
			{posts.map((post) => {
				return (
					<CardPost
						key={post.id}
						image={post.urls.full}
						altImage={post.alt_description}
						avatarImage={post.user.profile_image.small}
						avatarName={post.user.instagram_username}
						likes={post.likes}
						isRowDisplay
					/>
				)
			})}
		</Box>
	)
}
