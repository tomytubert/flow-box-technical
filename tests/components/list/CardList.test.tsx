import React from 'react'
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react'
import { CardList } from '../../../src/components/list/ImageList'

describe('Testing <ImageList />', () => {
	const post = {
		image: 'https://images.unsplash.com/photo-1634170380000-0b3b3b3b3b3b',
		altImage: 'A beautiful image',
		likes: 100,
		id: '1',
	}

	test('should match the snapshot', () => {
		const { container } = render(<CardList {...post} />)
		expect(container).toMatchSnapshot()
	})

	test('should render the image', () => {
		render(<CardList {...post} />)
		const { src, alt } = screen.getByRole('img') as HTMLImageElement
		expect(src).toBe(post.image)
		expect(alt).toBe(post.altImage)
	})

	test('should render the likes', () => {
		render(<CardList {...post} />)
		const likes = screen.getByText(post.likes)
        expect(likes).toBeTruthy()
    })

	test('should render the description', () => {
		render(<CardList {...post} />)
        const description = screen.getByText(post.altImage)
        expect(description).toBeTruthy()
        expect(description).toHaveTextContent(post.altImage);

	})
})
