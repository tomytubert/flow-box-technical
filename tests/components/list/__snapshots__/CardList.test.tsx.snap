// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ImageList /> should match the snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1oqv2c0-MuiPaper-root-MuiCard-root"
  >
    <img
      alt="A beautiful image"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-12qezwy-MuiCardMedia-root"
      height="194"
      loading="lazy"
      src="https://images.unsplash.com/photo-1634170380000-0b3b3b3b3b3b"
      width="200"
    />
    <div
      class="MuiCardContent-root css-147w5jl-MuiCardContent-root"
    >
      <div
        class="MuiBox-root css-syu8eu"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
          data-testid="FavoriteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          100
        </p>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 capitalize-first-letter css-r40f8v-MuiTypography-root"
      >
        A beautiful image
      </p>
    </div>
  </div>
</div>
`;
